aea/cli/scaffold.py

```diff
@@ -225,6 +220,16 @@ def scaffold_item(ctx: Context, item_type: str, item_name: str) -> None:
         else:
             fingerprint_item(ctx, item_type, new_public_id)
 
+        package_hash = IPFSHashOnly().hash_directory(str(dest))
+        new_public_id_with_hash = PublicId(
+            author_name, item_name, DEFAULT_VERSION, package_hash
+        )
+        if not to_local_registry:
+            logger.debug(f"Registering the {item_type} into {DEFAULT_AEA_CONFIG_FILE}")
+            existing_ids.add(new_public_id_with_hash)
+            with open_file(os.path.join(ctx.cwd, DEFAULT_AEA_CONFIG_FILE), "w") as fp:
+                ctx.agent_loader.dump(ctx.agent_config, fp)
+
```

Comment:
This change was supposed to be on the base PR directly but I forgot to commit before checking out to this branch 😅. While scaffolding we need to use the new public id format to register the newly scaffolded component in `aea-config.yaml`, this allows us to do that.